optuna/trial/_trial.py

```diff
@@ -61,9 +61,16 @@ def _init_relative_params(self) -> None:
         self.relative_search_space = self.study.sampler.infer_relative_search_space(
             study, self._cached_frozen_trial
         )
-        self.relative_params = self.study.sampler.sample_relative(
-            study, self._cached_frozen_trial, self.relative_search_space
-        )
+        self._relative_params: Optional[Dict[str, Any]] = None
```

Comment:
Since `_init_relative_params()` does not actually initialize relative_params due to this change, how about removing this method and writing these logics directly in `__init__()` method?